### PR TITLE
Support auditing to a storage account in VNet.

### PR DESCRIPTION
**Pull request by bashahee**
## Description

This PR enables customers to define auditing settings such that the target of the audit records is a storage account that its access was limited to requests originating from specified IP addresses, IP ranges or from a list of subnets in an Azure Virtual Network (VNet).
No changes in interfaces of existing cmdlets, and no new cmdlets were added.
Two new tests were added, in addition to a new test records to existing tests.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by bashahee** [comment]
@msJinLei Hi, you assigned me and added `needs-revision,` but you did not write any comments or requests. Moreover, all checks have passed.

**Comment by msJinLei** [comment]
> @msJinLei Hi, you assigned me and added `needs-revision,` but you did not write any comments or requests. Moreover, all checks have passed.

image

File: src/Sql/Sql/Auditing/Cmdlet/SetAzSqlDatabaseAudit.cs [hunk 1]
```diff
@@ -108,6 +109,8 @@ public class SetAzSqlDatabaseAudit : SqlDatabaseAuditCmdlet
             HelpMessage = AuditingHelpMessages.PassThruHelpMessage)]
         public SwitchParameter PassThru { get; set; }
 
+        public Guid RoleAssignmentId { get; set; } = default(Guid);
```
[the remark this answers is not in the document]
Comment (bashahee) [reply src/Sql/Sql/Auditing/Cmdlet/SetAzSqlDatabaseAudit.cs:112]: This parameter is a test hook for the two tests I added, same thing was done in https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/RoleAssignments/NewAzureRoleAssignmentCommand.cs.

I'll explain. The new added code assigns a role for the given storage, to do so it generates a role assignment id, the generated id is part of one of the request URIs that are being recorded when running the tests in record mode. Afterwards, tests fail to run in playback mode,  as a new id is being generated in each run, and the corresponding request URI is not being found.

This way, the test supplies the value of the id, the code does not generate it, and we succeed running the tests in playback mode.

**Comment by bashahee** [comment]
@msJinLei Can the PR be merged?

File: src/Sql/Sql/Auditing/Cmdlet/SetAzSqlDatabaseAudit.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (msJinLei) [reply src/Sql/Sql/Auditing/Cmdlet/SetAzSqlDatabaseAudit.cs:112]: OK

**Pull request merged**